crates/csharp/src/function.rs

```diff
@@ -225,6 +225,136 @@ impl<'a, 'b> FunctionBindgen<'a, 'b> {
 
         results.push(lifted);
     }
+
+    fn handle_result_import(&mut self, operands: &mut Vec<String>) {
+        if self.interface_gen.csharp_gen.opts.with_wit_results {
+            uwriteln!(self.src, "return {};", operands[0]);
+            return;
+        }
+
+        let mut payload_is_void = false;
+        let mut previous = operands[0].clone();
+        let mut vars: Vec<(String, Option<String>)> = Vec::with_capacity(self.results.len());
+        if let Direction::Import = self.interface_gen.direction {
+            for ty in &self.results {
+                let tmp = self.locals.tmp("tmp");
+                uwrite!(
+                    self.src,
+                    "\
+                                        if ({previous}.IsOk) {{
```

Comment:
Nit: Would you mind adjusting the indentation for this and the other `uwrite` expressions that follow?  E.g. put the `if` under the `"` on the previous line or similar and indent the `var` on the next line four columns past that, etc.  It's a bit jarring to see it indented way to the right.